shigeifinder/1.3.2/Dockerfile

```diff
@@ -0,0 +1,139 @@
+    tar -xjf samtools-${SAMTOOLS_VER}.tar.bz2 && \
+    rm samtools-${SAMTOOLS_VER}.tar.bz2 && \
+    cd samtools-${SAMTOOLS_VER} && \
+    ./configure && \
+    make && \
+    make install
+
+# install ShigEiFinder; make /data
+RUN wget https://github.com/LanLab/ShigEiFinder/archive/refs/tags/v${SHIGEIFINDER_VER}.tar.gz && \
+    tar -xvf v${SHIGEIFINDER_VER}.tar.gz && \
+    rm v${SHIGEIFINDER_VER}.tar.gz && \
+    cd ShigEiFinder-${SHIGEIFINDER_VER} && \
+    python3 setup.py install && \
+    mkdir /data
+
+# final working directory is /data
+WORKDIR /data
+
+# set locale settings for singularity compatibility. Set PATH to include bwa
+ENV LC_ALL=C \
+    PATH="${PATH}:/bwa/bwa-${BWA_VER}"
+
+# test layer
+FROM app as test
+
+ARG SHIGEIFINDER_VER
+
+# install ncbi datasets tool (pre-compiled binary); place in $PATH
+RUN wget https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets && \
+    chmod +x datasets && \
+    mv -v datasets /usr/local/bin
+
+# check dependencies and print help options
+RUN shigeifinder --check && \
+    shigeifinder --help
+
+# insert test here for downloading a couple of representative genomes (FASTA) and run them through ShigEiFinder
+# Shigella sonnei genome: https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_019947675.1/
+# BioSample: SAMN21386344
+ARG GENBANK_ACCESSION="GCA_019947675.1"
+RUN datasets download genome accession ${GENBANK_ACCESSION} --filename ${GENBANK_ACCESSION}.zip && \
+    mkdir -v ${GENBANK_ACCESSION}-download && \
+    unzip ${GENBANK_ACCESSION}.zip -d ${GENBANK_ACCESSION}-download && \
+    rm ${GENBANK_ACCESSION}.zip && \
+    mv -v ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}*.fna ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}.genomic.fna && \
+    shigeifinder -i ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}.genomic.fna \
+      --hits \
+      -t 2 \
+      --output shigeifinder.${GENBANK_ACCESSION}.out && \
+    head -n 2 shigeifinder.${GENBANK_ACCESSION}.out
+
+# Shigella flexneri 2a str. 301
+# https://www.ncbi.nlm.nih.gov/data-hub/taxonomy/198214/
+ARG GENBANK_ACCESSION="GCF_000006925.2"
+RUN datasets download genome accession ${GENBANK_ACCESSION} --filename ${GENBANK_ACCESSION}.zip && \
+    mkdir -v ${GENBANK_ACCESSION}-download && \
+    unzip ${GENBANK_ACCESSION}.zip -d ${GENBANK_ACCESSION}-download && \
+    rm ${GENBANK_ACCESSION}.zip && \
+    mv -v ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}*.fna ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}.genomic.fna && \
+    shigeifinder -i ${GENBANK_ACCESSION}-download/ncbi_dataset/data/${GENBANK_ACCESSION}/${GENBANK_ACCESSION}.genomic.fna \
+      --hits \
+      -t 2 \
+      --output shigeifinder.${GENBANK_ACCESSION}.out && \
+    head -n 2 shigeifinder.${GENBANK_ACCESSION}.out
```

Comment:
I think this could be in a for loop, which I think is a personal choice rather than something I think you should change

Reply:
you are absolutely right, I was just racing to get the image made so I can add it to our workflow. Perhaps for the next version I'll implement the for loop. Sometimes they are trick to get working right in dockerfiles since `/bin/sh` is used by default instead of `/bin/bash`